src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,197 @@
 ---
 title: 'k6 Extensions'
-redirect: 'https://k6.io/blog/extending-k6-with-xk6'
----
\ No newline at end of file
+---
+
+Traditionally, extending k6 with custom functionality that isn't available in the
+open source tool has been possible in one of two ways:
+
+- By importing a JavaScript library. This is a simple way to extend the functionality
+  of a test script, though it has some drawbacks. Since JavaScript in k6 runs in a
+  virtual machine [that is unlike the one used in web browsers and
+  NodeJS](/#what-k6-does-not), it's not possible to use libraries that write to
+  disk, use JS APIs other than the ones built into k6, or need some new binary
+  protocol support. Furthermore, while k6 performs well in most use cases, all
+  execution is interpreted by the [Go JavaScript runtime](https://github.com/dop251/goja),
+  which can impact the performance of certain resource intensive operations
+  (cryptography, compression, etc.).
+
+- By forking the [k6 repository](https://github.com/k6io/k6), doing the changes in Go
+  and submitting a pull request for review by the core team (preferably following
+  the [contribution guidelines](https://github.com/k6io/k6/blob/master/CONTRIBUTING.md)).
+  This is a great way to contribute new core features, but it can be a lengthy
+  process, and submissions might be rejected if they don't align with the long-term
+  vision of the project.
+
+To address these issues and allow the community to more easily adapt k6 to fit their
+needs, [k6 v0.29.0](https://github.com/k6io/k6/releases/tag/v0.29.0) introduced
+the [xk6 framework](https://github.com/k6io/xk6) and the concept of k6 extensions.
+
+
+## What are k6 extensions?
+
+k6 extensions are standalone Go projects that call k6 APIs but are otherwise
+unrestricted in their functionality. This provides the freedom for extension authors
```

Comment:
In the beginning, we should focus on describing the benefits of this feature. 

Having the ability to use extensions to experiment with features that can become part of the feature is a "secondary" benefit, imho.

Reply:
This is mostly a definition paragraph, but what other benefits would you mention?

The historical introduction explains the issues extensions are meant to solve for end users, so I'm not sure what else to mention here.

Reply:
Also about https://github.com/grafana/k6-docs/pull/346/files#r661994675

My view here is that the intro and the `what is k6 extension` and `what is xk6` sections are a little verbose.  For the Intro sections, I rather going more direct to the point in Docs and extend the historical in the blog post.

I like `Extension Types` section, but perhaps the `Getting Started` section could be better connected with the intro.

The article is good. If you want, I can give it a try in the future for a new proposal of the intro.